core/src/main/java/org/apache/brooklyn/core/objs/AbstractEntityAdjunct.java

```diff
@@ -421,6 +421,7 @@ public void setDisplayName(String name) {
     public void setEntity(EntityLocal entity) {
         if (destroyed.get()) throw new IllegalStateException("Cannot set entity on a destroyed entity adjunct");
         this.entity = entity;
+        this.execution = ((EntityInternal) entity).getExecutionContext();
```

Comment:
The problem was that `execution` is *NEVER* set otherwise, and it's used in `getContext()` that's what.